src/test/java/jenkins/advancedqueue/jobinclusion/strategy/JobInclusionJobPropertyTest.java

```diff
@@ -3,53 +3,238 @@
+import hudson.util.StreamTaskListener;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import jenkins.advancedqueue.PriorityConfiguration;
+import org.junit.After;
 import org.junit.Before;
+import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TestName;
 import org.jvnet.hudson.test.JenkinsRule;
 
 public class JobInclusionJobPropertyTest {
-    private JobInclusionJobProperty jobProperty;
-    private JobInclusionJobProperty.DescriptorImpl descriptor;
+
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
 
     @Rule
-    public JenkinsRule jenkinsRule = new JenkinsRule();
+    public TestName testName = new TestName();
+
+    private JobInclusionJobProperty property;
+    private FreeStyleProject project;
+    private FreeStyleBuild build;
+    private StreamBuildListener listener;
+    private Launcher launcher;
 
     @Before
-    public void setup() {
-        jobProperty = new JobInclusionJobProperty(true, "TestJobGroup");
-        descriptor = new JobInclusionJobProperty.DescriptorImpl();
+    public void setUp() throws Exception {
+        property = new JobInclusionJobProperty(true, "testGroup");
+        project = j.createFreeStyleProject("testFolder_" + testName.getMethodName());
+        build = j.buildAndAssertSuccess(project);
+        listener = new StreamBuildListener(new PrintStream(System.out), StandardCharsets.UTF_8);
+        launcher = new hudson.Launcher.LocalLauncher(StreamTaskListener.fromStdout());
+    }
+
+    @After
+    public void deleteProject() throws Exception {
+        project.delete();
+    }
+
+    @Test
+    public void getDescriptor() {
+        JobInclusionJobProperty.DescriptorImpl descriptor = property.getDescriptor();
+        assertNotNull(descriptor);
+        assertEquals("XXX", descriptor.getDisplayName());
+        assertTrue(descriptor instanceof JobInclusionJobProperty.DescriptorImpl);
+    }
+
+    @Test
+    public void getJobActions() {
+        // Setup: Add a job group to the PriorityConfiguration
+        PriorityConfiguration.get().getJobGroups().clear(); // Clear existing job groups
+        PropertyBasedJobInclusionStrategy.addJobGroup("testJobGroup", "testGroup");
+
+        // Retrieve job actions
+        ListBoxModel jobActions = PropertyBasedJobInclusionStrategy.getJobGroups();
+        assertNotNull(jobActions);
+        assertFalse(jobActions.isEmpty());
+
+        // Check the size of the list
+        assertEquals("Expected number of job actions", 1, jobActions.size());
+
+        // Check the properties of the first element
+        ListBoxModel.Option firstOption = jobActions.get(0);
+        assertNotNull(firstOption);
+        assertEquals("Expected display name", "testJobGroup", firstOption.name);
+        assertEquals("Expected value", "testGroup", firstOption.value);
     }
 
     @Test
-    public void getJobGroupNameTest() {
```

Comment:
Please don't delete the existing tests (like `getJobGroupNameTest`).  Your pull request should add to the tests rather than replacing existing tests.